scripts/scil_tractogram_resample.py

```diff
@@ -56,29 +63,33 @@ def _build_arg_parser():
 
     p.add_argument('--never_upsample', action='store_true',
                    help='Make sure to never upsample a tractogram.\n'
-                        'Useful when downsample batch of files using bash.')
+                        'Useful when downsampling a batch of files using '
+                        'bash.')
 
     # For upsampling:
-    upsampling_group = p.add_argument_group('Upsampling params')
-    upsampling_group.add_argument('--point_wise_std', type=float, default=1,
-                                  help='Noise to add to existing streamlines '
-                                       'points to generate new ones [%(default)s].')
-    upsampling_group.add_argument('--tube_radius', type=float, default=1,
-                                  help='Maximum distance to generate streamlines '
-                                       'around the original ones [%(default)s].')
-    upsampling_group.add_argument('--gaussian', metavar='SIGMA', type=int,
-                                  help='Sigma for smoothing. Use the value of '
-                                       'surrounding X,Y,Z points on the '
-                                       'streamline to blur the streamlines.\n'
-                                       'A good sigma choice would be around 5.')
-    upsampling_group.add_argument('-e', dest='error_rate', type=float, default=0.1,
-                                  help='Maximum compression distance in mm '
-                                       '[%(default)s].')
-
-    upsampling_group.add_argument('--keep_invalid_streamlines',
-                                  action='store_true',
-                                  help='Keep invalid newly generated streamlines'
-                                       ' that may go out of the \nbounding box.')
+    up_group = p.add_argument_group('Upsampling params')
+    up_group.add_argument('--point_wise_std',
+                          type=ranged_type(float, 0, None, min_excluded=True),
+                          default=1,
+                          help='Noise to add to existing streamlines '
+                               'points to generate new ones [%(default)s].')
+    up_group.add_argument('--tube_radius',
```

Comment:
For that particular case, isn't `, type=float, nargs='?', const=1,` enough. Like in `scil_tractogram_compute_density_map.py`
If you don't use it, its none
If you use it and type nothing its 1.0
If you use it and type something its your value

Reply:
I could, but the default was already 1 before my changes. What do you want the default to be if you are upsampling? To not do it, or keep as is; to do it with 1?

(Same question for point_wise_std)

Reply:
If you are upsampling it still has to be set if the user wants otherwise we don't use tube_radius.

For point_wise_std I guess it has to be default 1, otherwise without this or default 0 what the streamlines will look like ?

Reply:
We can remove the default too, but I think the user needs a range of values that make sense (for tube radius)
A value is too small for tube_radius and point_wise_std and the streamline is identical and the tractogram will look weird (like all streamlines are duplicated exactly next to each other) and too big and you make streamline go through regions they should not.

I think the default was 1-1 just to show in the help that the order of magnitude (and in millimeter) user should use.

Reply:
Ok, so, if I understand, I could:
1) remove both defaults and set them to const
2) if we are upsampling, verify that at least one value has been set between point wise and tube 
3) verify that the values are in a normal range, else print a warning

What could be "normal" ranges for both values?

Reply:
(I am pushing changes now with a warning above 10. I can change again if needed)

Reply:
That's the behavior I would expect, we can revise when Maxime try it if its not intuitive, but for now I think that perfect.